diff --git a/gettext-tools/gnulib-lib/execute.c b/gettext-tools/gnulib-lib/execute.c
index ff6f506..f583fbe 100644
--- a/gettext-tools/gnulib-lib/execute.c
+++ b/gettext-tools/gnulib-lib/execute.c
@@ -156,16 +156,16 @@ execute (const char *progname,
        copy of the environment block - ignoring the effects of putenv() and
        [un]setenv().  */
     {
-      exitcode = spawnvpe (P_WAIT, prog_path, (const char **) prog_argv,
-                           (const char **) environ);
+      exitcode = spawnvpe (P_WAIT, prog_path, (char* const*) prog_argv,
+                           (char* const*) environ);
       if (exitcode < 0 && errno == ENOEXEC)
         {
           /* prog is not a native executable.  Try to execute it as a
              shell script.  Note that prepare_spawn() has already prepended
              a hidden element "sh.exe" to prog_argv.  */
           --prog_argv;
-          exitcode = spawnvpe (P_WAIT, prog_argv[0], (const char **) prog_argv,
-                               (const char **) environ);
+          exitcode = spawnvpe (P_WAIT, prog_argv[0], (char* const*) prog_argv,
+                               (char* const*) environ);
         }
     }
   if (nulloutfd >= 0)
diff --git a/gettext-tools/gnulib-lib/spawn-pipe.c b/gettext-tools/gnulib-lib/spawn-pipe.c
index 006fa8b..d21d3f8 100644
--- a/gettext-tools/gnulib-lib/spawn-pipe.c
+++ b/gettext-tools/gnulib-lib/spawn-pipe.c
@@ -198,16 +198,16 @@ create_pipe (const char *progname,
        copy of the environment block - ignoring the effects of putenv() and
        [un]setenv().  */
     {
-      child = spawnvpe (P_NOWAIT, prog_path, (const char **) prog_argv,
-                        (const char **) environ);
+      child = spawnvpe (P_NOWAIT, prog_path, (char* const*) prog_argv,
+                        (char* const*) environ);
       if (child < 0 && errno == ENOEXEC)
         {
           /* prog is not a native executable.  Try to execute it as a
              shell script.  Note that prepare_spawn() has already prepended
              a hidden element "sh.exe" to prog_argv.  */
           --prog_argv;
-          child = spawnvpe (P_NOWAIT, prog_argv[0], (const char **) prog_argv,
-                            (const char **) environ);
+          child = spawnvpe (P_NOWAIT, prog_argv[0], (char* const*) prog_argv,
+                            (char* const*) environ);
         }
     }
   if (child == -1)
